### ISSUE TITLE
Restructuring ui and actors module

### ISSUE THREAD

**Issue by skoudoro** [fury-gl/fury#64]
## Description

`ui.py` and `actors.py` start to be huge so it would good to start some thinking on this file restructuration. We need to convert them to package and submodule. 

It would be good to do a short meeting (30-45min) this week or next week to talk about it. If you have some suggestion, feel free to describe your proposition here.

What do you think @MarcCote @ranveeraggarwal @Garyfallidis @dmreagan? Thank you in advance for your feedback.

**Comment by ranveeraggarwal** [comment]
I see a few approaches here:

1. 2D and 3D elements in separate (two) files. But as the elements grow, we'll be back to the same issue.
2. Separate file for each top level elements (the derivative of an element can join the same file). The advantage here is that filenames will document the range of elements we have, but this might suffer from the problem of file bloat in general.
3. Categorize the elements and have separate files for each category. For instance, FileMenu (and future menus) can fall in the menus file while the ring and the line sliders can be in `sliders.py`. 

Furthermore, is there a nice way to do something like forward declaration of classes in Python? Or perhaps a file-level docstring, so that one doesn't have to go through the entire file to get the range of features we provide?

**Comment by MarcCote** [comment]
I thought I had already replied to this issue. My bad :(.

I'm all for splitting the huge `ui.py` into multiple files. I don't have a strong opinion about the separation but shorter files are always better IMO.

@ranveeraggarwal about forward declarations, you mean like listing the available classes/components in the `__init__.py` for instance?

**Comment by ranveeraggarwal** [comment]
@MarcCote yeah importing the available APIs in `__init.py__` sounds right. That would also ensure that we maintain the flat-ness of the UI API (`import x from ui` instead of `import x from ui.component_name`).
